### PR TITLE
Match Pokemon LeafGreen 1.0

### PR DESCRIPTION
**Pull request by PikalaxALT**
- [x] Code alignment
- [x] Scripts alignment
- [x] Data alignment
- [x] Graphics alignment
- [x] Code matching
- [x] Scripts matching
- [x] Data matching
- [x] Graphics matching

### CODE REVIEW

File: src/slot_machine.c [hunk 1]
```diff
@@ -213,9 +213,15 @@ static const u16 gUnknown_8464994[] = INCBIN_U16("graphics/slot_machine/unk_8464
 static const u16 gUnknown_84649B4[] = INCBIN_U16("graphics/slot_machine/unk_84649b4.gbapal");
 static const u16 gUnknown_84649D4[] = INCBIN_U16("graphics/slot_machine/unk_84649d4.gbapal");
 static const u16 gUnknown_84649F4[] = INCBIN_U16("graphics/slot_machine/unk_84649f4.gbapal");
+#if defined(FIRERED)
 static const u32 gUnknown_8464A14[] = INCBIN_U32("graphics/slot_machine/unk_8464a14.4bpp.lz");
-static const u16 gUnknown_846504C[] = INCBIN_U16("graphics/slot_machine/unk_846504c.gbapal");
```
Comment (jiangzhengwenjz) [inline src/slot_machine.c]: is unk_846504c.pal deleted?

File: src/title_screen.c [hunk 2]
```diff
@@ -222,26 +272,36 @@ static void (*const sSceneFuncs[])(s16 * data) = {
...
     {(const void *)sBlankFireSpriteTiles, 0x500, 1},
     {(const void *)gGraphics_TitleScreen_BlankObjTiles, 0x400, 2},
     {(const void *)sSlashSpriteTiles, 0x800, 3}
 };
+#elif defined(LEAFGREEN)
+static const struct CompressedSpriteSheet sSpriteSheets[] = {
+    {(const void *)sLeafSpriteTiles, 0x580, 0},
+    {(const void *)sBlankFireSpriteTiles, 0x100, 1},
+    {(const void *)gGraphics_TitleScreen_BlankObjTiles, 0x400, 2},
+    {(const void *)sSlashSpriteTiles, 0x800, 3}
```
Comment (jiangzhengwenjz) [inline src/title_screen.c]: Is there any problem in defining them as u32 arrays?

File: data/graphics.s [hunk 3]
```diff
@@ -16532,15 +16551,46 @@ gGraphics_TitleScreen_CharizardMap:: @ 8EADEE4
 
 gGraphics_TitleScreen_BackgroundPals:: @ 8EAE094
 	.incbin "graphics/title_screen/unk_8eae094.gbapal"
+	.else
+	.ifdef LEAFGREEN
+gGraphics_TitleScreen_PokemonFireRedLogoPals:: @ 8EAB6C4
```
Comment (jiangzhengwenjz) [inline data/graphics.s]: These symbols should be renamed if they're serving both versions

File: src/slot_machine.c (hunk 1 above)
Comment (PikalaxALT) [reply src/slot_machine.c]: No but it should be. The palette now sources from the PNG.

**Pull request merged**